### ISSUE TITLE
Expose mesh structures in DB through API

### ISSUE THREAD

**Issue by y0sher** [spacemeshos/api#129]
# Motivation

It would be super useful to be able to look at the mesh at the consensus level and see how every miner acted in the past, it's not easy to read anomalies of 50 blocks pointing to 50 blocks in every layer and more, but with such visualization we might be able to recognize faulty nodes whether because of bugs or adversarial nodes.. 

## Task 

Basically we should expose an API endpoint to read layers, epochs and blocks from the database, since we can only care about 
data inside the database this might not add a lot of synchronization complexity. 
this should return all the data we have about these objects, for example for a block we would want the header as well with the view, votes and the active set size.

**Comment by avive** [comment]
We need to figure out what is the appropriate api service to add these too. If this block data is part of the mesh which is in consensus and supposed to be identical across all nodes after tortoise verified the block's layer then it should go into the mesh api service. If this data is different from node to node then it should go into the node service which is designed to display data as the node sees it which may be different than the state of other nodes. So, it will be helpful if we can list the data we want to add and it is possible that some data is node-specific and some is mesh- wide. 
Also - are these data structures you mention above stable in the node and ready for api integration? @y0sher

**Comment by y0sher** [comment]
Actually they aren't final yet, they are about to change when the verifying tortoise will be merged. This data is mesh data, should or consistent across nodes (maybe not for recent layers).

**Comment by lrettig** [comment]
Layers, epochs, blocks, etc. are already exposed through the API - I think there are two questions to ask here:

1. What's missing? E.g., we don't show things like edges/votes right now
2. Does the interface need to be different for a specific use case, e.g., debugging or analysis? Right now you can only see blocks via layers, maybe we'd want a different endpoint to dump all block data, or to fetch a block by block ID, that sort of thing

**Comment by y0sher** [comment]
for debugging and analysis, this is a real time view to see how miners behave, with blockchains and their explorers usually a block is created every few minutes and it's pretty easy to see its content and the previous block which he points to, this way you can see how every miner behaves.. with mesh this is a little bit trickier because even if you have access to all this information ins some structural format it will still be hard to follow. and yet I think this could be a very powerful debugging tool. 

basically what's missing could be just adding view and vote edges in block's info (textually) and linking to the block pointed and it's layer, we could probably show more interesting info like the identity that created this block and other block she created..

**Comment by lrettig** [comment]
Makes sense. Some of this stuff is already there, e.g., smesher that created a block, but adding the other stuff would be nice for debugging. I don't see any real harm in adding this to the existing endpoints consumed by the explorer/dash backend, they can just ignore data items they don't care about. Or we could think about designing a new, more detailed debug endpoint if we think the use cases are really different.
